### DISCUSSION TITLE
add new user's credentials with creating new session.

### DISCUSSION THREAD

**Discussion by Ansh-Rathod** [supabase/supabase#11665] [Questions] [answered]
I wanted to add a new user in the authentication section without creating a new session.
i used this exact same example.

https://supabase.com/docs/reference/javascript/auth-admin-createuser

```
          const { error, data } = await supabase.auth.admin.createUser({
            email: formData.email,
            password: formData.password,
            user_metadata: {
              name: formData.firstName + " " + formData.lastName,
            },
          });
```
I was getting this error.

<img width="321" alt="Screenshot 2023-01-14 at 6 51 20 PM" src="https://user-images.githubusercontent.com/67627096/212473857-e238adcf-34a1-4a64-97b8-a0cf2e8014f3.png">

**Comment by Ansh-Rathod** [comment]
my db.ts file from im exporting supabase/

```
import { createClient } from "@supabase/supabase-js";

const supabase = createClient(
  "https://xxxxxx.supabase.co",
  "",
  {
    auth: {
      autoRefreshToken: true,
      persistSession: true,
    },
  }
);

export default supabase;

```

**Comment by GaryAustin1** [comment, marked as answer]
This should be done server side only and you need to use service role key for that calll.

**Reply by Ansh-Rathod** [reply]
thank you issue is solved
